### PR TITLE
nixos/munin: New options (and some bugfixes) for service configuration

### PR DESCRIPTION
**Pull request by ToxicFrog**
###### Motivation for this change

The old Munin configuration had some outright bugs (wrong fonts used for graph generation, documentation links out of date) and was missing a lot of useful knobs. These changes fix those bugs, and add options for plugin configuration, loading extra plugins, disabling builtin plugins, and applying custom styles to the HTML that Munin generates.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ToxicFrog** [comment]
On further testing, the extraAutoPlugins option doesn't work reliably due to the way Munin searches for plugins. I'll see about fixing that this evening.

File: nixos/modules/services/monitoring/munin.nix
Comment (florianjacob) [inline nixos/modules/services/monitoring/munin.nix]: How about
```
      ${lib.optionalString (cronCfg.extraCSS != "") "staticdir ${customStaticDir"}}
```
- should give the same result and be more explicit.

**Comment by florianjacob** [comment]
`nixos/tests/munin.nix` disables a plugin via `ignore_file ^apc_nis$`, it would make sense to switch that over to the new module option. Regarding previous `ignore_file` way vs. deleting the plugins like your option does, will that make any difference for munin? There is no other way for plugins to be enabled than linking them to `/etc/munin/plugins`, right?

Overall, this gives the munin the necessary portion of care, and shrinks my `extraConfig` to 1/8th. Did not test `extraAutoPlugins`, though.

File: nixos/modules/services/monitoring/munin.nix
Comment (ToxicFrog) [reply nixos/modules/services/monitoring/munin.nix]: Done.

**Comment by ToxicFrog** [comment]
I've updated tests/munin.nix to use the new disabledPlugins setting, and it seems to work.

**Comment by florianjacob** [comment]
@GrahamcOfBorg test munin

**Comment by florianjacob** [comment]
@GrahamcOfBorg build nixosTests.munin

**Comment by florianjacob** [comment]
@GrahamcOfBorg build nixosTests.munin

**Comment by ToxicFrog** [comment]
Made a small tweak to the CSS commit (so that the example code doesn't look hideous on warning/critical graphs) and added a new commit to bump the Munin version to the latest stable release (and fix a small bug in the build).

**Comment by FRidh** [comment]
@GrahamcOfBorg build nixosTests.munin

**Comment by ToxicFrog** [comment]
Is there anything else I need to do here (sync and resolve conflicts?) or is it just waiting for a maintainer to merge it?

**Comment by florianjacob** [comment]
@FRidh do you have time for a review, or can recommend someone else?

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by Infinisil** [comment]
Alright I think it's fine to stay. When my other comment is addressed and the merge conflict fixed this is fine to merge.

File: nixos/modules/services/monitoring/munin.nix
Comment (ToxicFrog) [reply nixos/modules/services/monitoring/munin.nix]: `cp` won't work here because the goal is to append it to the existing CSS files, since it's not meant to completely replace the built-in CSS, just (potentially) override selected rules. I've replaced `passAsFile` and `cat` with appropriate use of `echo >>` instead.

**Review by infinisil — APPROVED** [review]

**Comment by ToxicFrog** [comment]
Rebased against latest master, conflicts resolved.

**Comment by domenkozar** [comment]
@GrahamcOfBorg build tests.munin

**Comment by Infinisil** [comment]
@GrahamcOfBorg test munin

**Pull request merged**
